### PR TITLE
Use OpenSearch rather than ElasticSearch in Gradle builds

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

This PR includes changes which replace ElasticSearch with OpenSearch in our Gradle builds. Additionally, I updated Java code so that it does not reference ElasticSearch. Some links remain.

This PR should remove all reliance on ElasticSearch dependencies. We are still pulling in `org.elasticsearch:jna`, but OpenSearch is the project pulling in this dependency.

There is more work to be done on this, but I want to make smaller PRs for it.
 
### Issues Resolved

None.
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by chenqi0805 — APPROVED** [review]

File: examples/trace-analytics-sample-app/sample-app/analytics-service/build.gradle [hunk 1]
```diff
@@ -27,9 +27,9 @@ repositories {
 dependencies {
     implementation('org.springframework.boot:spring-boot-starter-web')
     testImplementation('org.springframework.boot:spring-boot-starter-test')
-    implementation 'org.elasticsearch.client:elasticsearch-rest-high-level-client:7.8.1'
-    implementation 'org.elasticsearch.client:elasticsearch-rest-client:7.8.1'
-    implementation 'org.elasticsearch:elasticsearch:7.8.1'
+    implementation 'org.opensearch.client:opensearch-rest-high-level-client:1.1.0'
+    implementation 'org.opensearch.client:opensearch-rest-client:1.1.0'
+    implementation 'org.opensearch:opensearch:1.1.0'
```
Comment (chenqi0805) [inline examples/trace-analytics-sample-app/sample-app/analytics-service/build.gradle:32]: I think we are not even using these dependency in the analytics-service sample app now. Those depedency was for the legacy logic in loggingController: https://github.com/chenqi0805/simple-ingest-transformation-utility-pipeline-1/blob/enh/kibana-trace-analytics-plugin/examples/odfe-pipes-trace-app/analytics-service/src/main/java/com/example/restservice/LoggingController.java
which has been significantly simplified.

We could either remove them in this PR or in later PR. Also you might want to pay attention to application.properties in the resources:

```
server.port=8087
elasticsearch.host=elasticsearch
elasticsearch.port=9200
```

elasticsearch.host=elasticsearch, elasticsearch.port=9200 can both be removed

Comment (laneholloway) [reply examples/trace-analytics-sample-app/sample-app/analytics-service/build.gradle:32]: @dlvenable Can you check if these are quick fixes? If so, let's fix them in this PR.

Comment (dlvenable) [reply examples/trace-analytics-sample-app/sample-app/analytics-service/build.gradle:32]: Yes, I'll fix them in this PR. Thanks for catching this.

I think we are not using it as well, but I wanted to remove these dependencies from our whole build.gradle dependency chain.

Comment (dlvenable) [reply examples/trace-analytics-sample-app/sample-app/analytics-service/build.gradle]: Thanks @chenqi0805 , I have removed those dependencies and the properties from the application.properties file.

File: data-prepper-plugins/otel-trace-group-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/oteltracegroup/OTelTraceGroupPrepperTests.java [hunk 2]
```diff
@@ -238,7 +238,7 @@ public void testTraceGroupFieldAlreadyPopulated() throws IOException {
     @Test
     public void testTraceGroupProcessMultiWorker() throws IOException, ExecutionException, InterruptedException {
         /*
-         * Note: we only test the threadsafety of the business logic in OtelTraceGroupPrepper. The elasticsearch REST client
+         * Note: we only test the threadsafety of the business logic in OtelTraceGroupPrepper. The OpenSearch REST client
          * itself is thread-safe {https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_changing_the_client_8217_s_initialization_code.html}.
```
Comment (laneholloway) [inline data-prepper-plugins/otel-trace-group-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/oteltracegroup/OTelTraceGroupPrepperTests.java:242]: what is the OpenSearch link?

**Pull request merged**